hedera-node/docs/design/services/smart-contract-service/frictionless-airdrops-system-contracts.md

```diff
@@ -0,0 +1,166 @@
+| `0xc1acfe5d`           | `function cancelAirdrops(PendingAirdrop[] memory pendingAirdrops) external returns (int64 responseCode)`            | TokenCancel      | `ResponseCode` | The response code from the call |
+| `0x241b6af7`           | `function claimAirdrops(PendingAirdrop[] memory pendingAirdrops) external returns (int64 responseCode)`             | TokenClaim       | `ResponseCode` | The response code from the call |
+| `0x012ea0b1`           | `function rejectTokens(address[] memory ftAddresses,  NftId[] memory nftIds) external returns (int64 responseCode)` | TokenReject      | `ResponseCode` | The response code from the call |
+
+New system contract functions must be added to a new `IHRC904` interface to support airdropping tokens.
+
+| Function Selector Hash |                                                Function Signature                                                | HAPI Transaction | Responsible service |    Response    |                                 |
+|------------------------|------------------------------------------------------------------------------------------------------------------|------------------|---------------------|----------------|---------------------------------|
+| `0xcef5b705`           | `function cancelAirdropFT(address receiverAddress) external returns (uint256 responseCode)`                      | TokenCancel      | HTS                 | `ResponseCode` | The response code from the call |
+| `0xad4917cf`           | `function cancelAirdropNFT(address receiverAddress, int64 serialNumber) external returns (uint256 responseCode)` | TokenCancel      | HTS                 | `ResponseCode` | The response code from the call |
+| `0xa83bc5b2`           | `function claimAirdropFT(address senderAddress) external returns (uint256 responseCode)`                         | TokenClaim       | HTS                 | `ResponseCode` | The response code from the call |
+| `0x63ada5d7`           | `function claimAirdropNFT(address senderAddress, int64 serialNumber) external returns (uint256 responseCode)`    | TokenClaim       | HTS                 | `ResponseCode` | The response code from the call |
+| `0x76c6b391`           | `function rejectTokenFT() external returns (uint256 responseCode)`                                               | TokenReject      | HTS                 | `ResponseCode` | The response code from the call |
+| `0xa869c78a`           | `function rejectTokenNFTs(int64[] memory serialNumbers) external returns (uint256 responseCode)`                 | TokenReject      | HTS                 | `ResponseCode` | The response code from the call |
+| `0x966884d4`           | `function setAutomaticAssociations(boolean enableAutoAssociations) external returns (uint256 responseCode)`      | CryptoUpdate     | HAS                 | `ResponseCode` | The response code from the call |
```

Comment:
Q: should we be more explicit so that the function name signifies the binary nature of the set?
Maybe
```suggestion
| `0x966884d4`           | `function setUnlimitedAutomaticAssociations(boolean enableAutoAssociations) external returns (uint256 responseCode)`      | CryptoUpdate     | HAS                 | `ResponseCode` | The response code from the call |
```

Reply:
Agreed. This will improve the readability a lot.